bokehjs/src/lib/models/glyphs/multi_polygons.ts

```diff
@@ -128,6 +128,37 @@ export class MultiPolygonsView extends GlyphView {
     }
   }
 
+  protected _hit_rect(geometry: RectGeometry): Selection {
+    const {sx0, sx1, sy0, sy1} = geometry
+    const [x0, x1] = this.renderer.xscale.r_invert(sx0, sx1)
+    const [y0, y1] = this.renderer.yscale.r_invert(sy0, sy1)
+    const candidates = this.index.indices({x0, x1, y0, y1})
+    const hits = []
+    for (let i = 0, end = candidates.length; i < end; i++) {
+      const idx = candidates[i]
+      const sxss = this.sxs[idx]
+      const syss = this.sys[idx]
+      let hit = true
+      for (let j = 0, endj = sxss.length; j < endj; j++) {
+        for (let k = 0, endk = sxss[j][0].length; k < endk; k++) {
+          const sx = sxss[j][0][k]
+          const sy = syss[j][0][k]
+          if (!hittest.point_in_poly(sx, sy, [sx0, sx1, sx1, sx0], [sy0, sy0, sy1, sy1])) {
```

Comment:
Will the compiler life these constant lists out of the loop? If not, they would be redundantly constructed on every point of a polygon (possibly thousands of points for large polygons)

Reply:
Solid point might as well move them out to be sure.